`Required` rule: remove `SkipOnEmptyInterface` and add `emptyCallback` option (#342)
yiisoft/validator: README.md

#### Skipping empty values

By default, missing and empty values are validated (if the value is missing, it's considered `null`). That is

> undesirable if you need to allow not specifying a field. To change this behavior, use `skipOnEmpty: true`.

Reviewer: ```suggestion
undesirable if you need a field to be optional. To change this behavior, use `skipOnEmpty: true`.
```

> Note that not all rules have this option, but only those that implement `Yiisoft\Validator\SkipOnEmptyInterface`.
> For example, `Required` rule doesn't. For more details about its customization see "Requiring values" section.

Reviewer: ```suggestion
Note that not every rule has this option, but only the ones that implement `Yiisoft\Validator\SkipOnEmptyInterface`.
For example, `Required` rule doesn't. For more details see "Requiring values" section.
```

}
```

### Requiring values

> Use `Yiisoft\Validator\Rule\Required` rule to make sure value is presented. What values exactly are considered empty can
> be customized via `$emptyCallback` option. Normalization is not performed here, so only a callable or special class is 
> needed. For more details see "Skipping empty values" section.
> 
> ```php
> use Yiisoft\Validator\Rule\Required;
> use Yiisoft\Validator\SkipOnEmptyCallback\SkipOnNull;
> 
> $rules = [new Required(emptyCallback: new SkipOnNull())];
> ```

Reviewer: ```suggestion
Use `Yiisoft\Validator\Rule\Required` rule to make sure a value is present. What values are considered empty can
be customized via `$emptyCallback` option. Normalization is not performed, so only a callable or special class is 
needed. For more details see "Skipping empty values" section.

```php
use Yiisoft\Validator\Rule\Required;
use Yiisoft\Validator\SkipOnEmptyCallback\SkipOnNull;

$rules = [new Required(emptyCallback: new SkipOnNull())];
```
```
